tutorials/scripting/gdextension/gdextension_c_example.rst

```diff
@@ -0,0 +1,2165 @@
+    }
+
+    void gdexample_free_instance(void *p_class_userdata, GDExtensionClassInstancePtr p_instance)
+    {
+        if (p_instance == NULL)
+        {
+            return;
+        }
+        GDExample *self = (GDExample *)p_instance;
+        gdexample_class_destructor(self);
+        api.mem_free(self);
+    }
+
+When instantiating an object, first we create a new Sprite2D object, since
+that's the parent of our class. Then we allocate memory for our custom struct
+and call its constructor. We save the pointer to the Godot object in the struct
+as well like we mentioned earlier.
+
+Then we set our custom struct as the instance data. This will make Godot know
+that the object is an instance of our custom class and properly call our custom
+methods for instance, as well as passing this data back.
+
+Note that we return the Godot object we created, not our custom struct.
+
+For the ``gdextension_free_instance`` function, we just call the destructor and free the memory we
+allocated for the custom data. It is not necessary to destruct the Godot object
+since that will be taken care of by the engine itself.
+
+A demo project
+--------------
+
+Now that we can create and free our custom object, we should be able to try it
+out in an actual project. For this, you need to open Godot and create a new
+project on the ``demo`` folder. The project manager may warn you the folder
+isn't empty if you have compiled the extension before, you can safely ignore
+this warning this time.
+
+If you didn't compile the extension yet, it is the time to do it now. To do
+that, open a terminal or command prompt, navigate to the root folder of the
+extension and run ``scons``. It should compile quickly since the extension is
+very simple.
+
+Then, create a file called ``gdexample.gdextension`` inside the ``demo`` folder.
+This is a Godot resource that describes the extension, allowing the engine to
+properly load it. Put the following content in this file:
+
+.. code-block::
+
+    [configuration]
+
+    entry_symbol = "gdexample_library_init"
+    compatibility_minimum = "4.2"
+
+    [libraries]
+    macos.debug = "res://bin/libgdexample.dylib"
+    linux.debug = "res://bin/libgdexample.so"
+    windows.debug = "res://bin/libgdexample.dll"
+
+As you can see, ``gdexample_library_init`` is the same name of the function we
+defined in our ``init.c`` file. It is important that the names match because it
+is how Godot calls the entry point of the extension.
+
+We also set the compatibility minimum to 4.2, since we are targeting this
+version. It should still work on later versions. If you are using a later Godot
+version and rely on the new features, you need to increase this value to a
+version number that has everything you use.
+
+In the ``[libraries]`` section we set up the paths to the shared library on
+different platforms. Here there's only the debug versions since that's what we
+are working on for the example. Using :ref:`Feature tags <doc_feature_tags>` you
```

Comment:
If you're happy with the link showing the actual title of the page (in this case "Feature tags") you can omit the explicit link title:
```suggestion
are working on for the example. Using :ref:`doc_feature_tags` you
```
Alternately, you can use an explicit link title and make the link lowercase so its more natural in the context of a sentence:
```suggestion
are working on for the example. Using :ref:`feature tags <doc_feature_tags>` you
```

The current version seems worse than either option IMO